build/make-distribution.sh

```diff
@@ -218,22 +218,43 @@ function build_flink_client {
   cp "$PROJECT_DIR"/client-flink/flink-$FLINK_BINARY_VERSION-shaded/target/celeborn-client-flink-${FLINK_BINARY_VERSION}-shaded_$SCALA_VERSION-$VERSION.jar "$DIST_DIR/flink/"
 }
 
+function build_mr_client {
+  HADOOP_VERSION=$("$MVN" help:evaluate -Dexpression=hadoop.version $@ 2>/dev/null \
+      | grep -v "INFO" \
+      | grep -v "WARNING" \
+      | tail -n 1)
+  BUILD_COMMAND=("$MVN" clean package $MVN_DIST_OPT -pl :celeborn-client-mr-shaded-${HADOOP_VERSION} -am $@)
+
+    # Actually build the jar
+    echo -e "\nBuilding with..."
+    echo -e "\$ ${BUILD_COMMAND[@]}\n"
+
+    "${BUILD_COMMAND[@]}"
+
+    ## flink spark client jars
+    mkdir -p "$DIST_DIR/mr"
+    cp "$PROJECT_DIR"/client-mr/mr-shaded/target/celeborn-client-mr-shaded-${HADOOP_VERSION}-$VERSION.jar "$DIST_DIR/mr/"
+}
+
 if [ "$RELEASE" == "true" ]; then
   build_service
   build_spark_client -Pspark-2.4
   build_spark_client -Pspark-3.4
   build_flink_client -Pflink-1.14
   build_flink_client -Pflink-1.15
   build_flink_client -Pflink-1.17
+  build_mr_client mr
 else
   ## build release package on demand
   build_service $@
-  if [[ $@ != *"spark"* && $@ != *"flink"* ]]; then
+  if [[ $@ != *"spark"* && $@ != *"flink"* && $@ != *"mr"* ]]; then
       echo "Skip building client."
   elif [[ $@ == *"spark"* && $@ != *"flink"* ]]; then
     build_spark_client $@
   elif [[ $@ == *"flink"* && $@ != *"spark"* ]]; then
     build_flink_client $@
+  elif [[ $@ == *"mr"* ]]; then
```

Comment:
If we build with `-Pspark-3.4 -Pmr`, will only build spark client

Reply:
I'll add a note about Celeborn can not compile Spark or Flink or MR clients at the same time.